src/rdf4cpp/rdf/Serialize.hpp

```diff
@@ -0,0 +1,220 @@
+#ifndef RDF4CPP_SERIALIZE_HPP
+#define RDF4CPP_SERIALIZE_HPP
+
+#include <array>
+#include <cstdio>
+#include <cstring>
+#include <ostream>
+#include <string>
+#include <string_view>
+#include <system_error>
+
+namespace rdf4cpp::rdf {
+
+/**
+ * A cursor into a buffer
+ * Describes remaining length (size()) and current write head (data())
+ *
+ * Memory layout is guaranteed to be identical to:
+ * @code
+ * struct Cursor {
+ *      char *data;
+ *      size_t size;
+ * };
+ * @endcode
+ */
+struct Cursor {
```

Comment:
maybe directly call it BufferCurser?

Reply:
I think cursor is unambiguous, esp if we move this stuff into a subnamespace